Userland/Libraries/LibGfx/TextLayout.cpp

```diff
@@ -32,6 +32,18 @@ RefPtr<GlyphRun> shape_text(FloatPoint baseline_start, Utf8View string, Gfx::Fon
     Vector<Gfx::DrawGlyph> glyph_run;
     FloatPoint point = baseline_start;
     for (size_t i = 0; i < glyph_count; ++i) {
+        if (input_glyph_info[i].codepoint == '\t') {
+            auto ch_width = font.glyph_width('0');
+
+            auto tab_stop_dist = (ceil((x_offset / tab_width)) * tab_width) - x_offset;
+            if (tab_stop_dist < ch_width / 2) {
+                tab_stop_dist += tab_width;
+            }
```

Comment:
please clarify where this formula is coming from and what it does

Reply:
I hope this makes more sense after my explanation above that references one of the wpt tests.

Reply:
Ideally we'd have a spec quote that explains the formula, as a comment, just so it's clear to anyone reading. The relevant spec part seems to be step 2 of [this](https://drafts.csswg.org/css-text/#white-space-phase-2) which maybe we can more directly implement later.

Reply:
yeah good point! we can add a link and some comments :)